docs/src/main/sphinx/object-storage/file-system-s3.md

```diff
@@ -123,3 +123,138 @@ and secret keys, STS, or an IAM role:
 * - `s3.external-id`
   - External ID for the IAM role trust policy when connecting to S3.
 :::
+
+## Security mapping
+
+Trino supports flexible security mapping for S3, allowing for separate
+credentials or IAM roles for specific users or buckets/paths. The IAM role
+for a specific query can be selected from a list of allowed roles by providing
+it as an *extra credential*.
+
+Each security mapping entry may specify one or more match criteria. If multiple
+criteria are specified, all criteria must match. Available match criteria:
+
+- `user`: Regular expression to match against username. Example: `alice|bob`
```

Comment:
do we need to say what type of regex?

Reply:
And we should explain that the examples uses an OR and therefore matches for alice and bob

Reply:
Do we go into this level of detail in the other documentation? Note that this documentation is copied over from the original, so we could do a more major cleanup as a separate task.

Reply:
We should .. just have not gotten around to fixing that up.

Reply:
Let's follow-up with the docs improvement after this PR is merged.